src/wave.js

```diff
@@ -271,13 +274,55 @@ export class Wave extends mix(WaveBase).with(
         this.adjustOrbitControlsTarget(cellViewParams.center);
     }
 
+    collectAllAtoms() {
```

Comment:
Why exactluy did we need this function?

Reply:
This function is used because we have atoms in different groups and that function collects all atoms from this group. I use it when we rebuild all scene for creating new array of chosen atoms, I found an issue that on each state update we are rebuilding all scene that is not good for performance, I spoked with @tjduigna, and he said that it's a quite inefficient state managment, and he said that there are 2 possible solutions - 1. is full rewrite, and 2. is to refill atoms on each scene rebuild